inc/bans.php

```diff
@@ -113,20 +113,25 @@ static public function parse_range($mask) {
 		return array($ipstart, $ipend);
 	}
 	
-	static public function find($ip, $board = false, $get_mod_info = false) {
+	static public function find($ip, $board = false, $get_mod_info = false, $banid = false) {
 		global $config;
-		
+
+		if (!$banid)
+			$search = '(`ipstart` = :ip OR (:ip >= `ipstart` AND :ip <= `ipend`)))';
+		else
+			$search = '(``bans``.`id` = :ip))';
```

Comment:
this is not smart or good code, to overload `$ip` to hold two types, an int or an ip, switching on `$banid`. `$banid` should be NULL by default and be passed to the query as a param. the query should always bind it and decide internally to use or not depending on nullity.